pkg/ext/stores/kubeconfig/store.go

```diff
@@ -146,34 +147,40 @@ func New(wranglerContext *wrangler.Context, authorizer authorizer.Authorizer, us
 
 // EnsureNamespace ensures that the namespace for storing kubeconfig configMaps exists.
 func (s *Store) EnsureNamespace() error {
-	var backoff = wait.Backoff{
-		Duration: 100 * time.Millisecond,
-		Factor:   2,
-		Jitter:   .2,
-		Steps:    10, // Up to ~51 seconds.
-	}
-
-	return wait.ExponentialBackoff(backoff, func() (bool, error) {
-		_, err := s.nsCache.Get(namespace)
-		if err == nil {
-			return true, nil
+	var err error
+
+	s.ensureNamespace.Do(func() {
```

Comment:
I think (and inclined to) we can make the logic simpler and more resilient if we ditch sync.Once and just check with the namespace cache on every write. It should negligent time-wise in comparison to all the write operations.

Reply:
I'd like to avoid maintaining our own state and introducing a mutex that guards it.

Reply:
> Note further, this is common between kubeconfig and tokens. IMHO this could be placed into a common package.

Initially I was deliberately keeping them separate but in the end extracted EnsureNamespace.